program-runtime/src/invoke_context.rs

```diff
@@ -326,13 +326,14 @@ impl<'a> InvokeContext<'a> {
         instruction: &Instruction,
         signers: &[Pubkey],
     ) -> Result<(Vec<InstructionAccount>, Vec<IndexOfAccount>), InstructionError> {
-        // Finds the index of each account in the instruction by its pubkey.
-        // Then normalizes / unifies the privileges of duplicate accounts.
-        // Note: This is an O(n^2) algorithm,
-        // but performed on a very small slice and requires no heap allocations.
+        // We reference accounts by an u8 index, so we have a total of 256 accounts.
+        // This algorithm allocates the array on the stack for speed.
+        // On AArch64 in release mode, this function only consumes 912 bytes of stack.
+        let mut transaction_callee_map: [Option<u8>; 256] = [None; 256];
```

Comment:
I think it can be done without an `Option`. By initializing it to `u8::MAX` and using `index_in_callee.min(instruction_accounts.len() as u8)`.

Reply:
I did not use `u8::MAX`, because the `u8` index can theoretically index an account with a 255 index. There is no restriction to that in the monorepo.

Reply:
Also, for clarity, we better use `None`, than a `u8::MAX`, since the former explicitly means no account.